### ISSUE TITLE
Steam blocks ReviveInjector from working

### ISSUE THREAD

**Issue by brianjonespfk** [LibreVR/Revive#80]
The Steam version of Distance was just updated with official Rift 1.3 support.

I have tried multiple version of ReVive, but even when dragging Distance.exe onto ReVive.exe the game only loads in 2D mode. I've tried setting the game settings to Fullscreen and Windowed. I've tried running it with SteamVR open first, with SteamVR closed...etc. No matter what, the game just opens like normal in 2D mode on the monitor as if a VR device is not detected. If SteamVR is off when I try it, then the green light comes on on the headset but the Vive screen stays black (While the game loads up in 2D mode on the monitor). If SteamVR is open before launching Distance (dragging it onto Revive.exe) then it just opens the dashboard in the headset and shows that Distance is running, and I can use the controller to click "Exit game" but that is all. (And again it's up and running like normal on the monitor)

It's like it does not recognize there is a "Rift" and so it will not load the VR mode.

**Comment by brianjonespfk** [comment]
I also disabled "theater mode" for this game, otherwise if SteamVR is open before starting it it just opens it in the virtual theater mode.

**Comment by LibreVR** [comment]
Steam games are not yet supported. The Steam DRM blocks ReviveInjector from working, so Steam games will need another means of patching the game. If it's a 64-bit application, you could try to apply the Project Cars patch.

There are ways to fix ReviveInjector so it works with Steam, but it has the potential to get people banned from VAC if they're not careful. That's why I'm currently focusing on supporting Oculus Home games.

**Comment by Wibz** [comment]
Oddly, I have Distance working via steam / revive. The only issue I have is there is no audio at all.

**Comment by LibreVR** [comment]
That's very odd indeed, are you sure you're not just playing in theater mode?

**Comment by Wibz** [comment]
I don't think so,  though having not used theatre mode before, it's possible I was confused. But I'll double check when I can (probably about 3 hours after this post).

**Comment by Wibz** [comment]
It definitely is in VR mode, not theatre mode. Though there were a couple of oddities. When I tried to run it straight into VR it crashed very quickly. I then ran it in desktop mode from the .exe and it took about 30+ seconds to progress passed logging into steamworks. I then quit and launched it via revive and it worked fine in VR mode (without sound). Though when I exited the game, steam claimed I was still running Distance, but there was no Distance.exe running. I had to kill steam via taskmanager to clear it.

**Comment by LibreVR** [comment]
Sounds like you triggered a bug in Distance where Steam thinks it's already running and doesn't trigger the Steam launcher. This allows the ReviveInjector to inject into the game directly without Steam preventing it.

**Comment by Wibz** [comment]
I think you might be right. If I have time this evening I'll investigate a little more.

**Comment by LibreVR** [comment]
I don't think it's really worth trying to take advantage of this quirk though, doesn't seem very stable.

**Comment by Wibz** [comment]
The issues I reported were all unrelated. It just 'works' here. The exiting issue was a known issue related to Logitech. I can only assume they've not protected it with steam's DRM.

**Comment by brianjonespfk** [comment]
Ahh, I had no idea about the Steam limitation. The Steam version of Distance definitely has Steam's DRM. Thankfully I bought the game through humblebundle and there's a DRM free version I can download there. Will give that one a try!

**Comment by LibreVR** [comment]
That's interesting, it should use the default audio device. Possibly it uses an obscure variant of the Audio device functions that I haven't implemented correctly.

I'll see what I can do to fix that.

**Comment by Neo-One** [comment]
I bought TechnoLust on Steam, and I have no issues with it while using ReVive. I have imported the Injector.exe into Steam using LibreVR directions.

**Comment by LibreVR** [comment]
That means Technolust is likely not using the Steam DRM. You can notice this when starting the exectuable directly does not active that Steam loading dialog.

**Comment by Neo-One** [comment]
I see.

**Comment by LibreVR** [comment]
Oops, wanted to set a milestone on the missing sound issue specifically, but there's no separate issue report for it.

**Comment by Rojoyinc** [comment]
Don't buy on oculus store - don't support their  "PLATFORM DEPENDANT" concept.
